src/pyiem/util.py

```
@@ -547,19 +547,26 @@ def archive_fetch(
+    # triggering `RuntimeError: generator didn't stop after throw()`.
     try:
         resp = httpx.request(method.upper(), url, timeout=30)
         resp.raise_for_status()
-        if method.lower() == "head":
-            yield ""
-            return
-        with tempfile.NamedTemporaryFile(delete=False, suffix=suffix) as tmp:
-            tmp.write(resp.content)
-        yield tmp.name
-        return
     except Exception as exp:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)